### DISCUSSION TITLE
Custom client attributes/metadata

### DISCUSSION THREAD

**Discussion by starmage** [keycloak/keycloak#15135] [Q&A] [answered]
I have a need to have custom client attributes/metadata per client. What would be the best way of implementing this ?

The main requirement at this stage is to have some additional metadata/custom attributes per client - in our current use case, we need some extra information that we use to customize our UI per client (we are using 1 theme for a number of clients, but still want to customize the UI per client without the overhead of having a theme per client)

**Comment by sschu** [comment, marked as answer]
You can already add custom attributes to clients with the current API, see https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_clientrepresentation.
